packages/core-transaction-pool-redis/lib/connection.js

```diff
@@ -237,6 +245,51 @@ module.exports = class TransactionPool extends TransactionPoolInterface {
     return undefined
   }
 
+  /**
+   * Removes any transactions in the pool that have already been forged.
+   * @param  {Array} transactionIds
+   * @return {Array} IDs of pending transactions that have yet to be forged.
+   */
+  async removeForgedAndGetPending (transactionIds) {
+    const forgedIdsSet = new Set(await database.getForgedTransactionsIds(transactionIds))
+
+    await Promise.each(forgedIdsSet, async (transactionId) => {
```

Comment:
I would just use a simple `for (const transactionId of forgedIdsSet)` instead of adding another dependency in the form of `bluebird`.

Reply:
Yes we add another dep, but gain performance. Would go for this here as it can be time-consuming, e.g. wait for 150 trx in line or wait for 150  trx in parallel to be processed. And its called on couple of places.

Reply:
If you wish to run them in parallel you should use `Promise.all()` which is supported natively. `Promise.each()` is the same as a for loop.